client.go

```diff
@@ -364,6 +364,31 @@ func (c *Client) SetTagByLocator(locator string, tags *Tags) (*Tags, error) {
 	return tags, nil
 }
 
+func (c *Client) DownloadBuildLog(buildId int) (string, error) {
```

Comment:
Given the potential size of this return string in some cases and possibly wanting to avoid storing it all in memory, what do you think about this function taking an `io.Writer` instead of returning a string? That would allow users to pass in a `StringWriter` if that's all they wanted, to pipe directly to a file or implement their own Writer for e.g. filtering use-cases.

Reply:
Makes sense to me. Updated with an io.WriteCloser, because when using just an io.Writer it was non-trivial to detect in client code when the entire request body had been read.